### DISCUSSION TITLE
Large Map Pin collection performance issues

### DISCUSSION THREAD

**Discussion by faheys** [dotnet/maui#14778] [Q&A]
Has anyone successfully loaded a large number of Pins via the Map control's ItemSource property?

When the Pin collection is greater than a few hundred items there is a very noticible UI freeze while the pins are being rendered. Seems to take exponentially longer as the size of the Pin collection increases. For example the app freezes for nearly 1 minute with 2000 Pins.

**Comment by faheys** [comment]
I managed to workaround this issue by loading in the pin data within the view model constructor. This is almost instant when building in realease mode however it is not ideal for cases where large numbers of pins need to be dynamically loaded in.

**Comment by plppp2001** [comment]
@faheys which maui maps are you using?


I'm using this nuget:

https://github.com/themronion/Maui.GoogleMaps

**Reply by faheys** [reply]
I'm using the inbuilt Map control which uses Apple Maps on iOS and Google Maps on Android. My app is only targeting iOS for now though so can't speak on whether the issue exists on Android. 
I might look into doing some testing with the package you linked and see if it has better performance. Have you had any issues with performance using it?

**Reply by plppp2001** [reply]
> I'm using the inbuilt Map control which uses Apple Maps on iOS and Google Maps on Android. My app is only targeting iOS for now though so can't speak on whether the issue exists on Android. 
> I might look into doing some testing with the package you linked and see if it has better performance. Have you had any issues with performance using it?

I'm using this nuget, not the built in one...... Google maps on both ios and android.
Check out this video:

https://youtu.be/775rzp2U82M?feature=shared

**Reply by LeoJHarris** [reply]
Why are so many 3rd party components more reliable then MAUI component? Sometimes it seems like only one person is working on this library, we have 512 pins and running a stop watch it took 37 seconds to repopulate which I doubt customers will wait around for.
